Client/Frontend/UserContent/UserScripts/YoutubeAdblock.js

```diff
@@ -34,3 +35,127 @@
+        //   Support multiple trappers for the same property.
+        const trapProp = function(owner, prop, handler) {
+            if (handler.init(owner[prop]) === false) { return; }
+            const odesc = Object.getOwnPropertyDescriptor(owner, prop);
+            let prevGetter, prevSetter;
+            if (odesc instanceof Object) {
+                if (odesc.get instanceof Function) {
+                    prevGetter = odesc.get;
+                }
+                if (odesc.set instanceof Function) {
+                    prevSetter = odesc.set;
+                }
+            }
+            
+            Object.defineProperty(owner, prop, {
+                configurable: true,
+                get() {
+                    if (prevGetter !== undefined) {
+                        prevGetter();
+                    }
+                    return handler.getter();
+                },
+                set(a) {
+                    if (prevSetter !== undefined) {
+                        prevSetter(a);
+                    }
+                    handler.setter(a);
+                }
+            });
+        };
+        
+        const trapChain = function(owner, chain) {
+            const pos = chain.indexOf('.');
+            if (pos === -1) {
+                trapProp(owner, chain, {
+                    v: undefined,
+                    init: function(v) {
+                        if (mustAbort(v)) { return false; }
+                        this.v = v;
+                        return true;
+                    },
+                    getter: function() {
+                        return document.currentScript === thisScript
+                            ? this.v
+                            : cValue;
+                    },
+                    setter: function(a) {
+                        if (mustAbort(a) === false) { return; }
+                        cValue = a;
+                    }
+                });
+                return;
+            }
+            const prop = chain.slice(0, pos);
+            const v = owner[prop];
+            chain = chain.slice(pos + 1);
+            if (v instanceof Object || typeof v === 'object' && v !== null) {
+                trapChain(v, chain);
+                return;
+            }
+            trapProp(owner, prop, {
+                v: undefined,
+                init: function(v) {
+                    this.v = v;
+                    return true;
+                },
+                getter: function() {
+                    return this.v;
+                },
+                setter: function(a) {
+                    this.v = a;
+                    if (a instanceof Object) {
+                        trapChain(a, chain);
+                    }
+                }
+            });
+        };
+        
+        trapChain(window, chain);
+    };
+    
+    $<setJS>('ytInitialPlayerResponse.adPlacements', 'undefined');
+    $<setJS>('playerResponse.adPlacements', 'undefined');
```

Comment:
Looks good. The same approach won't necessarily work for other scriptlets, which could have non-string template parameters, but it should be fine here :+1: